reposync.py

```diff
@@ -216,13 +226,36 @@ def main():
         else:
             local_repo_path = opts.destdir + '/' + repo.id
 
+        # Ensure we don't traverse out of local_repo_path by dropping any
+        # packages whose remote_path is absolute or contains up-level
+        # references (unless explicitly allowed).
+        # See RHBZ#1600221 for details.
+        if not opts.allow_path_traversal:
+            newlist = []
+            skipped = False
+            for pkg in download_list:
+                if is_subpath(pkg.remote_path, local_repo_path):
+                    newlist.append(pkg)
+                    continue
+                my.logger.warning(
```

Comment:
A primary user of `../` paths is google compute engine's package repo. They have an odd yum repo structure where they use a `../../pool/` directory for packages similar to apt repos. We'd really like to mirror this repo into a directory of our choosing, so we hacked around it by running reposync with the `--urls` option and using wget to download those URLs into the correct directory structure. It'd be great if the reposync command could do the same instead of skipping these packages, I'm sure others have come across this issue.

Reply:
That's why we made it possible to enable the original (unsafe) behavior; just enable the `allow_path_traversal` configuration option.